src/js/components/auditlogs/eventdetails/terminalplayer.js

```diff
@@ -18,6 +18,128 @@ const searchAddon = new SearchAddon();
+      <script src="https://unpkg.com/xterm-addon-search@${search}/lib/xterm-addon-search.js"></script>
+      <script src="https://unpkg.com/xterm-addon-fit@${fit}/lib/xterm-addon-fit.js"></script>
+      <style type="text/css">
+        body {
+          display: grid;
+          justify-items: center;
+          max-width: 80vw;
+          margin: 10vh auto;
+          row-gap: 5vh;
+          font-family: 'Segoe UI', Roboto, Ubuntu, 'Helvetica Neue', Helvetica, Arial, sans-serif;
+        }
+        h2 {
+          color: #24444a;
+        }
+        button {
+          background-color: #921267;
+          padding: 1.3em 3.4em;
+          color: #fff;
+          font-weight: 700;
+          text-transform: uppercase;
+          border: 0;
+          border-radius: 3px;
+          cursor: pointer;
+        }
+        .disabled {
+          background-color: lightgrey;
+          opacity: 0.4;
+        }
+      </style>
+    </head>
+    <body>
+      <img src="https://vgy.me/0tXIM6.png" alt="mender-logo" />
+      <h2>Terminal playback</h2>
+      <div id="terminal"></div>
+      <div>
+        <button id="start" onclick="handleStart()">Start</button>
+        <button id="pause" class="disabled" disabled onclick="handlePause()">Pause</button>
+        <button id="stop" class="disabled" disabled onclick="handleStop()">Stop</button>
+      </div>
+      <script>
+        const byteArrayToString = body => String.fromCharCode(...body);
+        const transfer = '${JSON.stringify(content.map(item => ({ time: item.time, content: btoa(JSON.stringify(item.content)) })))}';
+        const content = JSON.parse(transfer);
+        let contentIndex = 0;
+        let timer;
+        const term = new Terminal();
+        const fitAddon = new FitAddon.FitAddon();
+        const searchAddon = new SearchAddon.SearchAddon();
+        term.loadAddon(searchAddon);
+        term.loadAddon(fitAddon);
+        term.open(document.getElementById('terminal'));
+        fitAddon.fit();
+        const startButton = document.getElementById('start');
+        const pauseButton = document.getElementById('pause');
+        const stopButton = document.getElementById('stop');
+
+        const resetPlayer = () => {
+          contentIndex = 0;
+          term.reset()
+          startButton.toggleAttribute('disabled');
+          pauseButton.toggleAttribute('disabled');
+          stopButton.toggleAttribute('disabled');
+          pauseButton.classList.toggle('disabled');
+          stopButton.classList.toggle('disabled');
+          startButton.classList.toggle('disabled');
+        }
+
+        const processContent = () => {
+          if (contentIndex === content.length) {
+            return resetPlayer();
+          }
+          const item = content[contentIndex];
+          contentIndex += 1;
+          let delay = 1;
+          if (item.content) {
+            const buffer = JSON.parse(atob(item.content));
+            term.write(byteArrayToString(buffer.data || []))
```

Comment:
during debugging I added console log here, but it did not work, it was next to impossible to debug, could you teach me how to debug here in case I need it? (and assume I only have the webdev tools from the browser and a text editor (vim).

Reply:
there you will only see the effects if you download the recording and open the html file - that is why this is in the `generateHtml` function... if you want to debug anything in the UI itself look a few lines below (around L195) and edit the `useEffect` function... 👍